### PR TITLE
Implement API + tests

### LINKED ISSUE #9: API for Profile Creation
Tasks:

- [x] Express API that updates the DB with data
- [x] Express API for getting data via `student_number`

Do work in the `testdb` database. The table is called `gamer_profile` in the schema `users_test`.

Users table looks like:

| Attribute/Column Name | Type |
| --------------------------- | ----- |
| first_name | charvar(50) |
| last_name | charvar(50) |
| <ins>student_number</ins> | charvar(8) |
| membership_tier | integer |
| banned | boolean |
| notes | charvar(250) |
| created_at | date |

### PR DESCRIPTION
**Pull request by joshhwuu**
Closes #9 

APIs for `GET`, `POST` and `DELETE` for the gamer profile database, we should consider renaming our db to something other than "testdb". Couple of follow-up issues for the future, we should merge this now for the sake of being able to start on frontend though. I went ahead and also started a test suite, done in mocha chai, test cases generated by LLM. Also added a GHA to run the tests, which warranted adding an identical table for running tests in, called gamer_profile_test, but should be empty unless running tests (each test cleans itself up).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,8 +2,23 @@
...
+    "mocha": "^10.7.3",
+    "prettier": "3.3.3",
+    "supertest": "^7.0.0"
+  },
+  "dependencies": {
+    "body-parser": "^1.20.3",
+    "dotenv": "^16.4.5",
+    "express": "^4.20.0",
+    "moment-timezone": "^0.5.45",
+    "pg": "^8.13.0"
```
Comment (Jaden51) [inline package.json:18]: We should leave this package.json for global node configurations. The tests folder for the API's should live in the `app` directory ie. `app/tests`. This way we don't have to add the node dependencies twice.

File: tests/test_api.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+describe("Gamer API", () => {
+  // Clean up the database before each test
+  beforeEach(async () => {
+    await db.query("DROP TABLE IF EXISTS users_test.gamer_profile_test");
+    await db.query(
+      "CREATE TABLE users_test.gamer_profile_test (LIKE users_test.gamer_profile INCLUDING ALL)",
+    );
+  });
+
```
Comment (Jaden51) [inline tests/test_api.js:14]: I'm unsure how mocha worths with express. Are you creating mocks for all the tests here or are you actually creating/edititing rows in the table?

File: app/app.js [hunk 3]
```diff
@@ -1,17 +1,148 @@
...
+ * @apiError {String} 404 Student not found.
+ * @apiError {String} 500 Server error.
+ */
+app.get("/gamer/:student_number", async (req, res) => {
+  const { student_number } = req.params;
+  try {
+    const result = await db.query(
+      "SELECT * FROM users_test.gamer_profile WHERE student_number = $1",
+      [student_number],
+    );
```
Comment (Jaden51) [inline app/app.js:39]: Move the actual query into a string variable

File: app/app.js [hunk 4]
```diff
@@ -1,17 +1,148 @@
...
+    if (result.rows.length === 0) {
+      return res.status(404).send("Student not found");
+    }
+    res.json(result.rows[0]);
+  } catch (err) {
+    res.status(500).send("Error finding gamer");
+  }
```
Comment (Jaden51) [inline app/app.js:46]: We should send back the `err` too

File: app/app.js [hunk 5]
```diff
@@ -1,17 +1,148 @@
...
+    notes,
+  } = req.body;
+  const created_at = moment().tz("America/Los_Angeles").format("YYYY-MM-DD");
+
   try {
-    const result = await db.query("SELECT current_database()");
-    res.send(result);
+    const result = await db.query(
+      `INSERT INTO users_test.gamer_profile 
+      (first_name, last_name, student_number, membership_tier, banned, notes, created_at)
```
Comment (Jaden51) [inline app/app.js:86]: Same as above, move the query into a separate string variable

File: app/app.js [hunk 6]
```diff
@@ -1,17 +1,148 @@
...
+        membership_tier,
+        banned,
+        notes,
+        created_at,
+      ],
+    );
+    res.status(201).send(result.rows[0]);
   } catch (err) {
-    console.error(err);
+    res.status(500).send("Error creating gamer");
```
Comment (Jaden51) [inline app/app.js:109]: Same as above

File: app/db.js [hunk 7]
```diff
@@ -1,11 +1,12 @@
...
-  user: "admin",
-  host: "54.245.134.32",
-  database: "postgres",
-  password: "password",
-  port: 5432,
+  user: process.env.DB_USER,
+  host: process.env.DB_HOST,
+  database: process.env.DB_NAME, // replace with name locally
+  password: process.env.DB_PASS, // replace with password locally
+  port: process.env.DB_PORT,
```
Comment (Jaden51) [inline app/db.js:9]: Do you have docs on how the env variables were setup?

File: tests/test_api.js (hunk 2 above)
Comment (joshhwuu) [reply tests/test_api.js:14]: Creates/edits rows to a test table  that lives in our database, not the live table where we'll have our real profiles

Comment (Jaden51) [reply tests/test_api.js:14]: Oh I see I read this wrong. So you create a table if it doens't exit and if it does it drops it? I'm good with that. I thought we were just creating a bunch of users/tables without ever cleaning it up.

File: package.json (hunk 1 above)
Comment (joshhwuu) [reply package.json:18]: Moved tests into app directory, running `npm test` from the root directory still requires all the dependencies for the CI, check out https://github.com/ubcesports/lounge-hub/actions/runs/10945602636/job/30390194369 failure.

**Review by Jaden51 — APPROVED** [review]
lgtm
